packages/lib/storybook/stories/wallets/AmazonPayExample.tsx

```diff
@@ -0,0 +1,132 @@
+
+        //URL selection
+        // http://localhost:3020/iframe.html?id=wallets-amazonpay--default&viewMode=story
+        // http://localhost:3020/?path=/story/wallets-amazonpay--default
+        // either has iframe or path
+
+        const urlSearchParams = new URLSearchParams(window.location.search);
+        const amazonCheckoutSessionId = urlSearchParams.get('amazonCheckoutSessionId');
+        const step = urlSearchParams.get('step');
+
+        // TODO move this to args
+        const chargeOptions = {
+            // chargePermissionType: 'Recurring',
+            // recurringMetadata: {
+            //     frequency: {
+            //         unit: 'Month',
+            //         value: '1'
+            //     }
+            // }
+        };
+
+        checkout.current = useSessions
+            ? await createSessionsCheckout({ showPayButton, countryCode, shopperLocale, amount })
+            : await createAdvancedFlowCheckout({
+                  showPayButton,
+                  countryCode,
+                  shopperLocale,
+                  amount
+              });
+
+        if (step === 'review') {
+            const amazonPayElement = new AmazonPay({
+                core: checkout.current,
+                ...chargeOptions,
+                /**
+                 * The merchant will receive the amazonCheckoutSessionId attached in the return URL.
+                 */
+                amazonCheckoutSessionId,
+                cancelUrl: 'http://localhost:3020/iframe.html?id=wallets-amazonpay--default&viewMode=story',
+                returnUrl: 'http://localhost:3020/iframe.html?id=wallets-amazonpay--default&viewMode=story&step=result'
+            });
+            setElement(amazonPayElement);
+        } else if (step === 'result') {
+            const amazonPayElement = new AmazonPay({
+                core: checkout.current,
+                /**
+                 * The merchant will receive the amazonCheckoutSessionId attached in the return URL.
+                 */
+                amazonCheckoutSessionId,
+                showOrderButton: false,
+                onSubmit: (state, component) => {
+                    return makePayment(state.data)
+                        .then(response => {
+                            if (response.action) {
+                                component.handleAction(response.action);
+                            } else if (response?.resultCode && checkPaymentResult(response.resultCode)) {
+                                alert(response.resultCode);
+                            } else {
+                                // Try handling the decline flow
+                                // This will redirect the shopper to select another payment method
+                                component.handleDeclineFlow();
+                            }
+                        })
+                        .catch(error => {
+                            throw Error(error);
+                        });
```

Comment:
@m1aw It is still using the old syntax for the `onSubmit`: handling action part with `component.handleAction`;  checking the resultCode manually instead of passing back to the library; I believe it also does not trigger the new callbacks onPaymentCompleted/onPaymentFailed. The goal is to make it work with the new syntax at least, so all payment methods follow the new flow.

Perhaps this needs an adjustment like it was done with GooglePay/ApplePay, as it seems that AmazonPay has an special flow to decline the payment (`component.handleDeclineFlow`)

Would that be possible?

Reply:
To the first part. I have changed it to use `actions.resolve` as part of the tests. I will pull those changes in.
On the second part, I have not made modifications to the way AmazonPay `onSubmit` works and also the way handleDeclineFlow works.

I thought that was the objective, not make more changes then the ones necessary here. I will take a look to also adjust it, I because I think I misunderstood it.

Reply:
Adjusting the callback to behave in the new way is a necessary one I'd say. We don't want merchants to have to implement two different submit flows depending on the payment method that they use.

Let's then merge this PR as it is blocking other tasks, and then you can iterate on this in a new PR along with the e2e tests.